ggshield/verticals/secret/extended_match.py

```diff
@@ -69,16 +69,37 @@ def from_match(
         cls, match: Match, lines: List[Line], is_patch: bool
     ) -> "ExtendedMatch":
         span = MatchSpan.from_match(match, lines, is_patch)
+
         start_line = lines[span.line_index_start]
         end_line = lines[span.line_index_end]
         line_index_start = start_line.pre_index or start_line.post_index
         line_index_end = end_line.pre_index or end_line.post_index
         assert line_index_start is not None and line_index_end is not None
         line_index_start += int(is_patch) - 1  # convert to 0-based
         line_index_end += int(is_patch) - 1
+
+        lines_with_secrets = [
+            lines[line_index]
+            for line_index in range(span.line_index_start, span.line_index_end + 1)
+        ]
+        stripped_match = match.match
+        if is_patch and len(lines_with_secrets) > 1:
+            # The secret is multi-line and in patch, we can have the first character to remove
+            stripped_match = "\n".join(
+                [
+                    (
+                        str_line_match[1:]
+                        if index > 0 and line.category != LineCategory.DATA
+                        else str_line_match
+                    )
+                    for index, (str_line_match, line) in enumerate(
+                        zip(match.match.split("\n"), lines_with_secrets)
+                    )
+                ]
```

Comment:
You can remove the `[`, `]` here: it's simpler and faster (no need to build a list)